src/tests/ftest/util/fio_utils.py

```diff
@@ -68,265 +75,365 @@ def __init__(self, namespace, command, test):
-            "--buffer_compress_chunk={}")
-        self.dedupe_percentage = FormattedParameter("--dedupe_percentage={}")
-        self.clat_percentiles = FormattedParameter("--clat_percentiles={}")
-        self.lat_percentiles = FormattedParameter("--lat_percentiles={}")
-        self.percentile_list = FormattedParameter("--percentile_list={}")
-        self.significant_figures = FormattedParameter(
-            "--significant_figures={}")
-        self.disk_util = FormattedParameter("--disk_util={}")
-        self.gtod_reduce = FormattedParameter("--gtod_reduce={}")
-        self.disable_lat = FormattedParameter("--disable_lat={}")
-        self.disable_clat = FormattedParameter("--disable_clat={}")
-        self.disable_slat = FormattedParameter("--disable_slat={}")
-        self.disable_bw_measurement = FormattedParameter(
-            "--disable_bw_measurement={}")
-        self.gtod_cpu = FormattedParameter("--gtod_cpu={}")
-        self.unified_rw_reporting = FormattedParameter(
-            "--unified_rw_reporting={}")
-        self.continue_on_error = FormattedParameter("--continue_on_error={}")
-        self.error_dump = FormattedParameter("--error_dump={}")
-        self.profile = FormattedParameter("--profile={}")
-        self.cgroup = FormattedParameter("--cgroup={}")
-        self.cgroup_nodelete = FormattedParameter("--cgroup_nodelete={}")
-        self.cgroup_weight = FormattedParameter("--cgroup_weight={}")
-        self.uid = FormattedParameter("--uid={}")
-        self.gid = FormattedParameter("--gid={}")
-        self.kb_base = FormattedParameter("--kb_base={}")
-        self.unit_base = FormattedParameter("--unit_base={}")
-        self.hugepage_size = FormattedParameter("--hugepage-size={}")
-        self.flow_id = FormattedParameter("--flow_id={}")
-        self.flow = FormattedParameter("--flow={}")
-        self.flow_watermark = FormattedParameter("--flow_watermark={}")
-        self.flow_sleep = FormattedParameter("--flow_sleep={}")
-        self.steadystate = FormattedParameter("--steadystate={}")
-        self.steadystate_duration = FormattedParameter(
-            "--steadystate_duration={}")
-        self.steadystate_ramp_time = FormattedParameter(
-            "--steadystate_ramp_time={}")
-
-        # fio api
+        # Middleware to use with fio.  Needs to be configured externally prior
+        # to calling run().
         self.api = BasicParameter(None, "POSIX")
 
-    def get_param_names(self):
-        """Get a sorted list of the defined FioCommand parameters."""
-        # Sort the FIO parameter names to generate consistent fio commands
-        all_param_names = super(FioCommand, self).get_param_names()
-        # move the 'name' option as the first option of fio job
-        all_param_names.remove("name")
-        all_param_names.insert(0, "name")
+        # List of fio job names to run
+        self.names = BasicParameter(None)
+        self._jobs = {}
```

Comment:
Should this be private? In cases where the user should need to access any of the attributes of the `FioJob` object individually having the 'jobs' dictionary private would prevent that flexibility. Maybe implementing `getter()` and `setters()` method on `FioJob` class would resolve that, but it might be tedious since there are so many attributes.

Reply:
You bring up an interesting point.  The original intent of make the list of `FioJob` objects (`self._jobs`) 'private' was to prevent the use from redefining the list items.  It does prevent the user from adjusting a `FioJob` object parameter value from a `FioCommand` object.  I could add a method that updates a `FioJob` `BasicParameter` object's value for a matching `FioJob` name.